lld/ELF/ScriptParser.cpp

```diff
@@ -1692,11 +1692,11 @@ ScriptParser::readSymbols() {
   while (!errorCount()) {
     if (consume("}"))
       break;
-    if (consumeLabel("local")) {
+    if (consume("local:") || (consume("local") && consume(":"))) {
       v = &locals;
       continue;
     }
-    if (consumeLabel("global")) {
+    if (consume("global:") || (consume("global") && consume(":"))) {
```

Comment:
We should check `local:` and `global:` below at `StringRef tok = next();`

Reply:
Sorry I am confused here why we need check `local:` and `global` below `StringRef tok = next();` ?  if the lexer didn't consume `global:`, `global : `, `local:`, `local : `, or `extern`, we do not need do other things for the symbol right? Please correct me if I understand here wrong. Thank you!

Reply:
* Manually inline the two `consumeLabel`  calls
* Move the expanded body to below, just above `next()`

Reply:
I committed the change but I think it's not what you were trying to say? If it is not could you please specify it more?